File: DKS-SPA/src/app/core/_models/s-dev-bom-file.ts
import { Pagination } from "./pagination";

export class SDevBomFile extends Pagination {

  factoryId: string;
  season: string;
  modelNo: string;
  modelName: string;
  article: string;
  team: string;

  userTeam:string;

  constructor() {

    super();
    this.isPaging = true; //開分頁
    
    this.factoryId = "";
    this.season = "";
    this.modelNo = "";
    this.modelName = "";
    this.article = "";
    this.team = "";
    this.userTeam = "N";
  }
  public setPagination(pagination: Pagination) {
    this.currentPage = pagination.currentPage;
    this.itemsPerPage = pagination.itemsPerPage;
    this.totalItems = pagination.totalItems;
    this.totalPages = pagination.totalPages;
  }
}

File: DKS-API/DTOs/SDevBomFile.cs
using System;
using DKS_API.Helpers;

namespace DKS_API.DTOs
{
    public class SDevBomFile : PaginationParams
    {
        public string FactoryId { get; set; }
        public string Season { get; set; }
        public string ModelNo { get; set; }
        public string ModelName { get; set; }
        public string Article { get; set; }
        public string Team { get; set; }
        public string UserTeam { get; set; }
        

    }
}